boat_simulator/common/sensors.py

```diff
@@ -0,0 +1,130 @@
+from dataclasses import dataclass
+
+from typing import Optional
+from numpy.typing import NDArray
+
+
+from boat_simulator.common.types import Scalar, ScalarOrArray
+
+from boat_simulator.common.generators import (
+    ConstantGenerator,
+    MVGaussianGenerator,
+    GaussianGenerator,
+)
+
+WindSensorGenerators = Optional[MVGaussianGenerator | ConstantGenerator]
+GPSGenerators = Optional[GaussianGenerator | ConstantGenerator]
+
+
+@dataclass
+class Sensor:
+    def update(self, **kwargs):
+        for attr_name, attr_val in kwargs.items():
+            if attr_name in self.__annotations__:
+                setattr(self, attr_name, attr_val)
+            else:
+                raise ValueError(
+                    f"""{attr_name} not a property in {self.__class__.__name__}
+                    expected {self.__annotations__}"""
+                )
+
+    def read(self, key):
+        if key in self.__annotations__:
+            return getattr(self, key)
+
+
+@dataclass
+class WindSensor(Sensor):
+    wind: ScalarOrArray
+    wind_noisemaker: WindSensorGenerators
+
+    @property  # type: ignore
+    def wind(self) -> ScalarOrArray:
+        return (
+            self._wind + self._wind_noisemaker.next()  # type: ignore
+            if self._wind_noisemaker is not None
+            else self._wind
+        )
+
+    @wind.setter
+    def wind(self, wind: ScalarOrArray):
+        self._wind = wind
+
+    @property  # type: ignore
+    def wind_noisemaker(self) -> WindSensorGenerators:
+        return self._wind_noisemaker
+
+    @wind_noisemaker.setter
+    def wind_noisemaker(self, noisemaker: WindSensorGenerators):
+        self._wind_noisemaker = noisemaker
```

Comment:
Given that we provided the option to have different types of noisemakers, I think it would be appropriate for a user to specify the noise generator, perhaps through a config file.

Reply:
I agree we should configure the noise maker by the configuration file, but I mean **while** the ROS node is running.

Reply:
No, that probably isn't supported.